From 2db02a2fbb1530c4ed8f6eb10e805bb2108ce0bb Mon Sep 17 00:00:00 2001
From: Ruchi Kandoi <kandoiruchi@google.com>
Date: Tue, 6 Nov 2018 16:16:17 -0800
Subject: [PATCH 09/10] Prevent Out of bound error in
 llcp_dlc_proc_rr_rnr_pdu()

Test: Nfc Enable/disable
Bug: 116788646
Change-Id: Iefa9ca0a291e0c59a442befccfaa2f17eecd461e
---
 src/nfc/llcp/llcp_dlc.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/nfc/llcp/llcp_dlc.c b/src/nfc/llcp/llcp_dlc.c
index d53f911..c17838d 100644
--- a/src/nfc/llcp/llcp_dlc.c
+++ b/src/nfc/llcp/llcp_dlc.c
@@ -1098,6 +1098,11 @@ static void llcp_dlc_proc_rr_rnr_pdu (UINT8 dsap, UINT8 ptype, UINT8 ssap, UINT1
     {
         error_flags = 0;
 
+        if (length == 0)
+        {
+            android_errorWriteLog(0x534e4554, "116788646");
+            return;
+        }
         rcv_seq = LLCP_GET_NR (*p_data);
 
         if (length != LLCP_PDU_RR_SIZE - LLCP_PDU_HEADER_SIZE)
-- 
2.7.4

